js/pro/poloniex.js

```diff
@@ -0,0 +1,886 @@
+                'watchTicker': true,
+                'watchTickers': true,
+                'watchTrades': true,
+                'watchBalance': true,
+                'watchStatus': false,
+                'watchOrders': true,
+                'watchMyTrades': false,
+            },
+            'urls': {
+                'api': {
+                    'ws': {
+                        'public': 'wss://ws.poloniex.com/ws/public',
+                        'private': 'wss://ws.poloniex.com/ws/private',
+                    },
+                },
+            },
+            'options': {
+                'tradesLimit': 1000,
+                'ordersLimit': 1000,
+                'OHLCVLimit': 1000,
+                'connectionsLimit': 2000, // 2000 public, 2000 private, 4000 total, only for subscribe events, unsubscribe not restricted
+                'requestsLimit': 500, // per second, only for subscribe events, unsubscribe not restricted
+                'channelToTimeframe': {
+                    'candles_minute_1': '1m',
+                    'candles_minute_5': '5m',
+                    'candles_minute_10': '10m',
+                    'candles_minute_15': '15m',
+                    'candles_minute_30': '30m',
+                    'candles_hour_1': '1h',
+                    'candles_hour_2': '2h',
+                    'candles_hour_4': '4h',
+                    'candles_hour_6': '6h',
+                    'candles_hour_12': '12h',
+                    'candles_day_1': '1d',
+                    'candles_day_3': '3d',
+                    'candles_week_1': '1w',
+                    'candles_month_1': '1m',
+                },
+            },
+        });
+    }
+
+    async authenticate () {
+        /**
+         * @ignore
+         * @method
+         * @description authenticates the user to access private web socket channels
+         * @see https://docs.poloniex.com/#authenticated-channels-market-data-authentication
+         * @returns {object} response from exchange
+         */
+        this.checkRequiredCredentials ();
+        const timestamp = this.milliseconds ();
+        const accessPath = '/ws/private';
+        const requestString = 'GET\n' + accessPath + '\nsignTimestamp=' + timestamp;
+        const url = this.urls['api']['ws']['private'];
+        const signature = this.hmac (this.encode (requestString), this.base64ToBinary (this.secret), 'sha256', 'base64');
+        const request = {
+            'event': 'subscribe',
+            'channel': [ 'auth' ],
+            'params': {
+                'key': this.apiKey,
+                'signTimestamp': timestamp,
+                'signature': signature,
+                'signatureMethod': 'HmacSHA256',  // optional
+                'signatureVersion': '2',          // optional
+            },
+        };
+        return await this.watch (url, accessPath, request, accessPath);
+    }
+
+    async subscribe (name, publicOrPrivate, symbols = undefined, params = {}) {
```

Comment:
`publicOrPrivate` looks like a boolean property I would probably call it something like `accessibility`